inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp

```diff
@@ -93,7 +93,7 @@ class LayerTestsCommon : public CommonTestUtils::TestsCommon {
 
             const auto max = std::max(CommonTestUtils::ie_abs(res), CommonTestUtils::ie_abs(ref));
             float diff = static_cast<float>(absoluteDifference) / static_cast<float>(max);
-            if (max == 0 || (diff > static_cast<float>(threshold))) {
+            if (std::isnan(static_cast<float>(res)) || max == 0 || (diff > static_cast<float>(threshold))) {
```

Comment:
1. Seems like it won't work for integer types. So I would suggest to have branching depending on data type.
2. Ii also might be worthwhile to check `ref` value as well.

Reply:
I changed for `std::isnan(static_cast<float>(absoluteDifference))`.
At the moment we can catch `nan` for `ref` and `res` (we shouldn't get `nan` in OV anyway). 

And I verified, for floating data type (float, double, bfloat16, float16) `std::isnan` throws `true`, for another cases `false`.
Thanks!